packages/desktop-client/src/components/autocomplete/CategoryAutocomplete.tsx

```diff
@@ -183,6 +208,39 @@ export function CategoryAutocomplete({
     [defaultCategoryGroups, categoryGroups, showSplitOption],
   );
 
+  const filterSuggestions = useMemo(
+    () =>
+      (
+        suggestions: (Suggestion | CategoryAutocompleteItem)[],
+        value: string,
+      ) => {
+        return suggestions
+          .filter((suggestion: Suggestion | CategoryAutocompleteItem) => {
+            const suggestionAsSuggestion = suggestion as Suggestion; // Cast to Suggestion
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏manual type coercions more often than not lead to subtle bugs down-the-line. Do we event need to use the `Suggestion` type here? AFAIK the `suggestions` (variable) underlying type is `CategoryAutocompleteItem`. So we could just use that.